server/src/test/java/org/apache/kafka/server/ClientMetricsManagerTest.java

```diff
@@ -299,8 +332,8 @@ public void testGetTelemetrySameClientImmediateRetryAfterPushFail() throws Unkno
         // Create new client metrics manager which simulates a new server as it will not have any
         // last request information but request should succeed as subscription id should match
         // the one with new client instance.
-
-        ClientMetricsManager newClientMetricsManager = new ClientMetricsManager(clientMetricsReceiverPlugin, 100, time);
+        kafkaMetrics = new Metrics();
```

Comment:
We create a new instance during setup already. Do we need to construct a new instance here? Ditto in a few other places.

Reply:
The reason of construction is that I do not want to associate earlier instance of kafkaMetrics to new client manager instance else the metrics will collide. Hence created the new one when another instance of client manger is created.

Reply:
Then, should we close the new `kafkaMetrics` at the end of the test? Also, could we make the new `kafkaMetrics` a local val? Otherwise, the old one won't be closed.  Finally, should we close `newClientMetricsManager` at the end of the test?

Reply:
Agree, done.